src/test/java/net/openhft/chronicle/threads/MediumEventLoopTest.java

```diff
@@ -0,0 +1,41 @@
+package net.openhft.chronicle.threads;
+
+import net.openhft.chronicle.core.threads.EventHandler;
+import org.junit.Test;
+
+import java.util.concurrent.BrokenBarrierException;
+import java.util.concurrent.CyclicBarrier;
+import java.util.stream.IntStream;
+
+import static org.junit.Assert.assertEquals;
+
+public class MediumEventLoopTest {
+
+    @Test
+    public void testAddingTwoEventHandlersBeforeStartingLoopIsThreadSafe() {
```

Comment:
can confirm that this fails for me (without the MediumEventLoop change) on Windows box 9 times out of 10